src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1931,17 +1931,21 @@ bool Lowering::IsValidConstForMovImm(GenTreeHWIntrinsic* node)
     GenTree* op1    = node->Op(1);
     GenTree* castOp = nullptr;
 
-    // TODO-Casts: why don't we fold the casts? MinOpts?
+    // In MinOpts, casts won't be folded during importation
     if (varTypeIsIntegral(node->GetSimdBaseType()) && op1->OperIs(GT_CAST))
```

Comment:
Does removing this handling entirely have any significant diffs outside MinOpts?

Reply:
I tried removing this, and (surprisingly?) I didn't get any diffs, so this code pattern doesn't seem common enough to even help MinOpts all that much. Maybe we should get rid of it altogether.